### PR TITLE
💻 Add ability to submit reviews

### LINKED ISSUE #175: Ability to add reviews
Popup (modal) to add reviews on button click.

Required:
- UI element to open modal
- Modal element
       - Comment
       - Ratings on multiple axes depending on type of review
       - Commenter information (program maybe?)
       - Probably a captcha.

### PR DESCRIPTION
**Pull request by evtyz**
# Purpose

Closes #175.

# Screenshots

image
image

# Libraries used

- [react-modal](https://www.npmjs.com/package/react-modal)

### CODE REVIEW

File: client/components/resource-details/SubmitReview.tsx [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+          onClick={onRequestClose}
+          xmlns="http://www.w3.org/2000/svg"
+          height="24px"
+          viewBox="0 0 24 24"
+          width="24px"
+          fill="#000000"
+        >
+          <path d="M0 0h24v24H0V0z" fill="none" />
+          <path d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z" />
+        </CloseButton>
```
Comment (jeffzh4ng) [inline client/components/resource-details/SubmitReview.tsx:297]: Can you extract the svg into it's own file in `/public?

File: client/components/resource-details/SubmitReview.tsx [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+          </Dropdown>
+        </DropdownRow>
+        <br />
+        <ButtonDiv>
+          <CancelButton type="button" onClick={onRequestClose}>
+            Cancel
+          </CancelButton>
+          <PostButton onClick={onRequestPost}>Post</PostButton>
+        </ButtonDiv>
+      </form>
```
Comment (jeffzh4ng) [inline client/components/resource-details/SubmitReview.tsx:355]: style: Can you add a newline between components?

File: client/components/resource-details/SubmitReview.tsx [hunk 3]
```diff
@@ -0,0 +1,403 @@
...
+  className,
+  name,
+  resourceSlug,
+}: {
+  className?: string
+  name: string
+  resourceSlug: string
+}) => {
+  const [modalIsOpen, setIsOpen] = useState(false)
+  const [formResponse, setFormResponse] = useState({})
```
Comment (jeffzh4ng) [inline client/components/resource-details/SubmitReview.tsx:373]: 👍

File: client/components/resource-details/SubmitReview.tsx [hunk 4]
```diff
@@ -0,0 +1,403 @@
...
+      </SubmitReviewBanner>
+      <Modal
+        isOpen={modalIsOpen}
+        onRequestClose={closeModal}
+        contentLabel="Example Modal"
+        style={ModalStyle}
+      >
+        <ModalContent onRequestClose={closeModal} resourceSlug={resourceSlug} />
+      </Modal>
+    </Fragment>
```
Comment (jeffzh4ng) [inline client/components/resource-details/SubmitReview.tsx]: same style nit: i like whitespace between components

File: client/components/resource-details/SubmitReview.tsx [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+
+const CancelButton = styled.button`
+  background: white;
+  border: medium solid #ff6961;
+  border-radius: 5px;
+  font-family: ${fontInter};
+  font-weight: ${fontWeight.bold};
+  color: #ff6961;
+  width: 100px;
+  padding: 5px 20px;
```
Comment (jeffzh4ng) [inline client/components/resource-details/SubmitReview.tsx:163]: @CarolinaLi and @lamvpham originally designed whitespace in multiples of `8px` (`4px`, `8px, `16px`, etc etc). Could you adjust spacing  hroughout this entire PR?

**Review by jeffzh4ng — CHANGES_REQUESTED** [review]

File: client/components/resource-details/SubmitReview.tsx [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+  const changeFaculty = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setAboutYou({ ...aboutYou, faculty: Faculty[e.target.value] })
+  }
+
+  const changeStatus = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setAboutYou({ ...aboutYou, status: Status[e.target.value] })
+  }
+
+  const onRequestPost = () => {
+    Object.keys(ratings).map((key) => (ratings[key] *= 10))
```
Comment (jeffzh4ng) [inline client/components/resource-details/SubmitReview.tsx:276]: We need to validate mandatory input here. Can address in the next PR though.

**Review by jeffzh4ng — DISMISSED** [review]

**Review by jeffzh4ng — DISMISSED** [review]
🚀🚀🚀

File: client/components/resource-details/SubmitReview.tsx (hunk 6 above)
Comment (evtyz) [reply client/components/resource-details/SubmitReview.tsx:277]: TBD in next PR

**Review by jeffzh4ng — APPROVED** [review]

**Pull request merged**